app/components/curriculum-inventory-report-header.js

```diff
@@ -17,16 +20,54 @@ const Validations = buildValidations({
 });
 
 export default Component.extend(Validations, ValidationErrorDisplay, {
+  flashMessages: service(),
+
   didReceiveAttrs(){
     this._super(...arguments);
     this.set('reportName', this.get('report.name'));
     this.set('isFinalized', this.get('report.isFinalized'));
+
+    // check if the d/l link points at the same domain as the current page is on.
+    let parser = document.createElement('a');
+    parser.href = this.get('report.absoluteFileUri');
+    let backendDomain = parser.hostname;
+    let frontendDomain = window.location.hostname;
+    this.set('downloadFromSameDomain', (backendDomain === frontendDomain));
```

Comment:
Putting all of this into the didReceiveAttrs() hook has gotten me into trouble in the past.  It doesn't seem to run every time.  Can you create computed properties for backendDomain, frontendDomain, and downloadFromSameDomain instead?  That will for sure update incase the report.absoluteFileUrl or the report itself gets modified.